### PR TITLE
Chore: Support request token

### PR DESCRIPTION
**Pull request by lucaswilliamgomes**

### CODE REVIEW

File: lib/incognia_api/api.rb [hunk 1]
```diff
@@ -15,8 +15,8 @@ def initialize(client_id:, client_secret:)
                                host: "https://api.incognia.com/api")
     end
 
-    def register_signup(installation_id:, address: nil, **opts)
-      params = { installation_id: installation_id }
+    def register_signup(address: nil, **opts)
```
Comment (lucaswilliamgomes) [inline lib/incognia_api/api.rb:18]: During this transition period, we would return the API error, like:
Screenshot from 2024-08-14 16-18-30
Wdyt?

File: spec/incognia_spec.rb
Comment (julianalucena) [inline spec/incognia_spec.rb:1]: You renamed everything, as a side-effect the tests do not guarantee that the `installation_id` use case is working. Do you agree?

File: lib/incognia_api/api.rb (hunk 1 above)
Comment (julianalucena) [reply lib/incognia_api/api.rb:18]: I think that's ok!

But I would let the method signature with `request_token` as a parameter, but with a default value (`nil`), to not break old usages. This way the lib is more elucidative on how we expect the client to use it.

The way it is now, it passes a wrong message to the client: you only need the account id for login and transactions, and for onboarding you don't need anything. Did you get my point?

**Review by julianalucena — CHANGES_REQUESTED** [review]

**Comment by marcelscr** [comment]
Mudflap is interested on using this feature :gogo:

File: spec/incognia_spec.rb
Comment (lucaswilliamgomes) [reply spec/incognia_spec.rb:1]: Yes, we just need to remember to remove these tests when the major release.
I did it [here](https://github.com/inloco/incognia-ruby/pull/28/commits/482fecf51b99dba016d9b27f54795212560fe37e).

File: lib/incognia_api/api.rb (hunk 1 above)
Comment (lucaswilliamgomes) [reply lib/incognia_api/api.rb]: Yes, I agree with you, and I did it [here](https://github.com/inloco/incognia-ruby/pull/28/commits/482fecf51b99dba016d9b27f54795212560fe37e).

File: spec/incognia_spec.rb
Comment (julianalucena) [reply spec/incognia_spec.rb:1]: Nice. Thanks.

File: lib/incognia_api/api.rb (hunk 1 above)
Comment (julianalucena) [reply lib/incognia_api/api.rb]: Nice, Lucs!

File: lib/incognia_api/api.rb [hunk 2]
```diff
@@ -15,8 +15,9 @@ def initialize(client_id:, client_secret:)
                                host: "https://api.incognia.com/api")
     end
 
-    def register_signup(installation_id:, address: nil, **opts)
-      params = { installation_id: installation_id }
+    def register_signup(request_token: nil, address: nil, **opts)
+      params = {}
+      params.merge!({ request_token: request_token }) if request_token
       params.merge!(opts)
       params.merge!(address&.to_hash) if address
```
Comment (julianalucena) [inline lib/incognia_api/api.rb:22]:
```suggestion
      params = { request_token: request_token }.compact
      params.merge!(opts)
      params.merge!(address&.to_hash) if address
```

**Review by julianalucena — COMMENTED** [review]
Some suggestions regarding received parameters that I think will make code more readable and will require less changes when `request_token` starts to be required.

**Review by julianalucena — APPROVED** [review]

**Pull request merged**